diff --git a/sim/Simif.cpp b/sim/Simif.cpp
index fff7151..8aa0620 100644
--- a/sim/Simif.cpp
+++ b/sim/Simif.cpp
@@ -140,8 +140,10 @@ void SpikeSimif::write_reg(int i, uint32_t data)
 }
 int SpikeSimif::cosim_instr(const Inst& inst)
 {
+#ifdef SPIKE_TRACE
     if (main_time > DEBUG_TIME)
         processor->set_debug(true);
+#endif
     uint32_t initialSpikePC = get_pc();
     uint32_t instSIM;
     bool fetchFault = 0;
diff --git a/sim/model_headers.h b/sim/model_headers.h
index 2fe2f29..5828f3b 100644
--- a/sim/model_headers.h
+++ b/sim/model_headers.h
@@ -1,31 +1,31 @@
-#include "VTop_TagBuffer.h"
-#include "VTop_CSR.h"
-#include "VTop_Rename__WC5.h"
-#include "VTop_IF_ICTable.h"
+#include "VTop__Syms.h"
+#include "VTop__Dpi.h"
 #include "VTop.h"
-#include "VTop_RegFile__W50_S20_N1_NB1.h"
-#include "VTop_ExternalAXISim.h"
-#include "VTop_RenameTable__ND5.h"
-#include "VTop_ROB.h"
+#include "VTop__pch.h"
+#include "VTop___024root.h"
 #include "VTop_Top.h"
-#include "VTop_IFetchPipeline.h"
+#include "VTop___024unit.h"
+#include "VTop_SoC.h"
+#include "VTop_ExternalAXISim.h"
+#include "VTop_Core.h"
 #include "VTop_MemRTL__W200_N40.h"
-#include "VTop_StoreQueue.h"
-#include "VTop___024root.h"
+#include "VTop_MemRTL1RW__W54_N40_WB15.h"
+#include "VTop_MemRTL1RW__W2_N40_WB2.h"
+#include "VTop_MemRTL__W200_N100_WB80.h"
 #include "VTop_IFetch.h"
+#include "VTop_ROB.h"
+#include "VTop_StoreQueue.h"
+#include "VTop_CSR.h"
+#include "VTop_Rename__WC5.h"
 #include "VTop_IF_Cache.h"
-#include "VTop__Dpi.h"
+#include "VTop_IF_CTable.h"
 #include "VTop_IF_MMIO.h"
-#include "VTop_MemRTL1RW__W54_N40_WB15.h"
+#include "VTop_TagBuffer.h"
+#include "VTop_IFetchPipeline.h"
 #include "VTop_BranchPredictor__N3.h"
-#include "VTop_RegFile__NB5_A1.h"
+#include "VTop_RenameTable__ND5.h"
+#include "VTop_IF_ICTable.h"
 #include "VTop_ReturnStack.h"
-#include "VTop_Core.h"
-#include "VTop_IF_ICache.h"
+#include "VTop_RegFile__NB5_A1.h"
 #include "VTop_RegFile__W23_S20_N3_NB1.h"
-#include "VTop_SoC.h"
-#include "VTop___024unit.h"
-#include "VTop_MemRTL__W200_N100_WB80.h"
-#include "VTop__Syms.h"
-#include "VTop__pch.h"
-#include "VTop_IF_CTable.h"
+#include "VTop_RegFile__W50_S20_N1_NB1.h"
